sdk/keyvault/azure-security-keyvault-keys/src/list_keys_single_page_result.cpp

```diff
@@ -125,3 +127,39 @@ DeletedKeySinglePage _detail::KeyPropertiesSinglePageSerializer::DeletedKeySingl
 
   return deletedKeySinglePage;
 }
+
+void DeletedKeySinglePage::OnNextPage(const Azure::Core::Context& context)
+{
+  // Before calling `OnNextPage` pagedResponse validates there is a next page, so we are sure
+  // NextPageToken is valid.
+  GetDeletedKeysSinglePageOptions options;
+  options.NextPageToken = NextPageToken;
+  *this = m_keyClient->GetDeletedKeysSinglePage(options, context);
+  CurrentPageToken = options.NextPageToken.Value();
+}
+
+void KeyPropertiesSinglePage::OnNextPage(const Azure::Core::Context& context)
+{
+  // Notes
+  // - Before calling `OnNextPage` pagedResponse validates there is a next page, so we are sure
+  // NextPageToken is valid.
+  // - KeyPropertiesSinglePage is used to list keys from a Key Vault and also to list the key
+  // versions from an specific key. When KeyPropertiesSinglePage is listing keys, the `m_keyName`
+  // fields will be empty, but for listing the key versions, the KeyPropertiesSinglePage needs to
+  // keep the name of the key in `m_kayName` because it is required to get more pages
```

Comment:
Nice. Thanks!
```suggestion
  // - KeyPropertiesSinglePage is used to list keys from a Key Vault and also to list the key
  // versions from a specific key. When KeyPropertiesSinglePage is listing keys, the `m_keyName`
  // fields will be empty, but for listing the key versions, the KeyPropertiesSinglePage needs to
  // keep the name of the key in `m_keyName` because it is required to get more pages.
```